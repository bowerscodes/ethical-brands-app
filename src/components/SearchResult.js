/* eslint-disable react/prop-types */
import React from "react";
import { Link } from "react-router-dom";
import "../styles/searchresult.css";
import { FaAngleLeft } from "react-icons/fa";
import animalIcon from "../assets/icons8-elephant-64.png";
import environmentIcon from "../assets/icons8-europe-64.png";
import labourIcon from "../assets/icons8-workers-64.png";
import philanthropyIcon from "../assets/icons8-donate-64.png";
import newsIcon from "../assets/icons8-magazine-64.png";

// eslint-disable-next-line react/prop-types
const SearchResult = ({
  logo,
  name,
  category,
  ownership,
  revenue,
  anTesting,
  anInfo,
  anScore,
  envInfo,
  envScore,
  labInfo,
  labScore,
  philDonate,
  philInfo,
  newsSource,
  newsHeadline,
  totalScore,
}) => {
  let scoreColor = "brand-score_"; // Conditional class naming based on score of brand
  if (totalScore >= 75) {
    scoreColor += "high";
  } else if (totalScore < 75 && totalScore > 49) {
    scoreColor += "average";
  } else {
    scoreColor += "low";
  }

  const formatting = Intl.NumberFormat("en-US", {
    // Format revenue number into currency type of USD
    style: "currency",
    currency: "USD",
  });
  const formattedRevenue = formatting.format(revenue);

  return (
    <div className="search-result_page">
      <div className="top-page">
        <Link to="/">
          <button type="submit" className="top-page_button">
            <FaAngleLeft className="top-page_arrow" />
          </button>
        </Link>
        <div className={`${scoreColor}`}>
          Score: <br /> {totalScore}
        </div>
      </div>

      <div className="search-results_summary">
        <img src={`${logo}`} alt="brand logo" className="search-results_logo" />
        <div>{`${name}, ${category}`}</div>
        <div>Ownership: {ownership}</div>
        <div>Revenue: {formattedRevenue}</div>
      </div>

      <div className="search-results-container">
        <div className="search-results_info">
          <div className="icon-and-score">
            <img src={animalIcon} alt="animal icon" className="icon" />
            {`${anScore}`}/5
          </div>
          Animal Testing: {`${anTesting}`}. <br /> {`${anInfo}`}
          {/* Empty div after each section to ensure there is space between when no information is provided */}
          <div />
        </div>

        <div className="search-results_info">
          <div className="icon-and-score">
            <img
              src={environmentIcon}
              alt="environment icon"
              className="icon"
            />
            {`${envScore}`}/5
          </div>
          {`${envInfo}`}
          <div />
        </div>

        <div className="search-results_info">
          <div className="icon-and-score">
            <img src={labourIcon} alt="labour icon" className="icon" />
            {`${labScore}`}/5
          </div>
          {`${labInfo}`}
          <div />
        </div>

        <div className="search-results_info">
          <div className="icon-and-score">
            <img
              src={philanthropyIcon}
              alt="philanthropy icon"
              className="icon"
            />
          </div>
          {`${philInfo}, ${philDonate}`}
          <div />
        </div>

        <div className="search-results_info">
          <div>
            <img src={newsIcon} alt="news icon" className="icon" />
          </div>
          <a href={newsSource} target="_blank" rel="noreferrer">
            {newsHeadline}
          </a>
          <div />
        </div>
      </div>
    </div>
  );
};

export default SearchResult;
