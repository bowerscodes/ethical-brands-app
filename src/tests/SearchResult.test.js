import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import SearchResult from "../components/SearchResult";

describe("search results", () => {
  it("snapshot test", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <SearchResult />
      </BrowserRouter>
    );
    const searchResults = asFragment();

    expect(searchResults).toMatchSnapshot();
  });
});
