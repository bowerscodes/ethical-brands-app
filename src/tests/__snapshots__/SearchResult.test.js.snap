// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search results snapshot test 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <a
      href="/"
    >
      <button
        class="search-results_button"
        type="submit"
      >
        <svg
          class="search-results_arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          />
        </svg>
      </button>
    </a>
    <div
      class="search-results_content"
    >
      <h3>
        Company Logo
      </h3>
      <div
        class="info"
      >
        <div>
          undefined, undefined
        </div>
        <div>
          Revenue: 
        </div>
        <div />
      </div>
      <div
        class="info-card"
      >
        <div>
          Animal Testing: undefined, undefined
        </div>
      </div>
      <div
        class="info-card"
      >
        <div>
          Environment: undefined, undefined
        </div>
      </div>
      <div
        class="info-card"
      >
        <div>
          Labour: undefined, undefined
        </div>
      </div>
      <div
        class="info-card"
      >
        <div>
          Philanthropy: undefined, undefined
        </div>
      </div>
      <div
        class="info-card"
      >
        <a>
          News: 
        </a>
      </div>
      <div>
        Score: 
      </div>
    </div>
  </div>
</DocumentFragment>
`;
